components/plurals/src/lib.rs

```diff
@@ -359,4 +338,39 @@ impl PluralRules {
     pub fn select<I: Into<PluralOperands>>(&self, input: I) -> PluralCategory {
         self.selector.select(&input.into())
     }
+
+    /// Lower level helper that allows extracting PluralRules-relevant data from a data provider
+    /// without constructing PluralRules
+    pub fn get_plural_data<'d, D: DataProvider<'d, provider::PluralRuleStringsV1<'d>> + ?Sized>(
```

Comment:
These are on PluralRules instead of DataProvider because we can still use slightly higher level types here.

I could instead move this to DataProvider and have it return PluralRuleStringsV1 if people prefer.

Reply:
I think I have a slight preference toward storing this on Provider data struct (and that's what I do on DTF) but that may be because I don't see the value you describe on storing it here yet.

Reply:
@zbraniecki if it's on Provider then I have to return PluralRuleStringsV1 instead of the converted type

Reply:
Oh, I don't think it should be on provider, more like https://github.com/unicode-org/icu4x/blob/main/components/datetime/src/provider/helpers.rs - on https://github.com/unicode-org/icu4x/blob/main/components/plurals/src/provider.rs#L22

Reply:
Ah. Should I be pulling out a PluralRulesV1 or a PluralRuleList?

I kinda prefer it on the type itself because it's all in one place then

Reply:
I'd advocate for having a function in the `icu_plurals::provider` module (probably in a sub module) that returns PluralRuleStringsV1.

Why PluralRuleStringsV1?

1. It's public
2. The same architecture (returning data structs) can be used in other crates
3. No dependencies outside of the `provider` module